lib/news_scraper/trainer/data_type.rb

```diff
@@ -0,0 +1,80 @@
+module NewsScraper
+  module Trainer
+    module DataType
+      class << self
+        def train(uri, payload, all_presets)
+          selected_presets = {}
+
+          NewsScraper::CLI.put_header(uri)
+          NewsScraper::CLI.log "Fetching information..."
+          NewsScraper::CLI.put_footer
+
+          data_types.each do |target_data_type|
+            data_type_presets = all_presets[target_data_type]
+            preset_results = preset_results(uri, payload, data_type_presets, target_data_type)
+
+            NewsScraper::CLI.put_header("Determining information for #{target_data_type}")
+            if preset_results.empty?
+              NewsScraper::CLI.log("No presets were found for #{target_data_type}. Skipping to next.")
+            else
+              selected_preset = select_preset(preset_results, data_type_presets, target_data_type)
+              selected_presets[target_data_type] = selected_preset if selected_preset
+            end
+            NewsScraper::CLI.put_footer
+
+            selected_presets[target_data_type] ||= { 'method' => "<<<<< TODO >>>>>", 'pattern' => "<<<<< TODO >>>>>" }
+          end
+
+          selected_presets
+        end
+
+        def preset_results(uri, payload, presets, data_type)
+          return {} unless presets
+
+          scrape_details = blank_scrape_details
+          presets.each_with_object({}) do |(preset_name, preset_details), hash|
+            scrape_details[data_type] = preset_details
+            train_transformer = Transformers::Article.new(
+              uri: uri,
+              payload: payload,
+              scrape_details: scrape_details
+            )
+
+            hash[preset_name] = train_transformer.transform[data_type.to_sym]
+          end.to_a
+        end
+
+        def blank_scrape_details
+          data_types.each_with_object({}) do |data_type, hash|
+            hash[data_type] = nil
+          end
+        end
+
+        def select_preset(preset_results, data_type_presets, target_data_type)
```

Comment:
This was the largest logic change - I just made the preset selecting a lot nicer and brought it to its own method